### PR TITLE
Adjust tx_power with asserted gain

### PR DESCRIPTION
**Pull request by madninja**

### CODE REVIEW

File: src/router/client.rs [hunk 1]
```diff
@@ -495,7 +496,7 @@ impl RouterClient {
         StateChannelMessage::offer(
             packet.packet().clone(),
             self.keypair.clone(),
-            self.region,
+            &self.region,
```
Comment (JayKickliter) [inline src/router/client.rs:499]: Minor nit: region isn’t Copy, so why this change?

Comment (madninja) [reply src/router/client.rs:499]: Because the offer call only needs a reference to it to construct the representing internal value

File: src/region.rs [hunk 2]
```diff
@@ -99,3 +101,56 @@ impl Region {
...
+        Ok(Self {
+            gain: value.gain,
+            params,
+            region,
+        })
+    }
+}
+
+impl RegionParams {
+    pub fn max_eirp(&self) -> Option<u32> {
```
Comment (JayKickliter) [inline src/region.rs:141]: A comment on units would be helpful. Deci-dBm?

Comment (JayKickliter) [reply src/region.rs:141]: Or maybe even a fixed-point type? I'd prefer to reduce the risk of someone down the line not taking into account a scaling factor.

Comment (madninja) [reply src/region.rs]: Added in [63f3de5](https://github.com/helium/gateway-rs/pull/262/commits/63f3de5bcdcd9e7cfdf0856edae28eabc98f2487)

**Review by JayKickliter — APPROVED** [review]
